test/e2e/cases/hcFromProbe.go

```diff
@@ -0,0 +1,89 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/cloudinterface"
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/loadbalancer"
+	utilsnamer "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/namer"
+)
+
+const (
+	svcNodePort = 30061
+	expectedHcPath = "/healthz"
+)
+
+func RunHCFromProbeTest() {
+	project, kubeConfigPath, lbName, ipName, clients := initDeps()
+	kubectlArgs := []string{"kubectl", fmt.Sprintf("--kubeconfig=%s", kubeConfigPath)}
+	setupHCFromProbe(kubectlArgs, ipName, clients)
+	defer func() {
+		cleanupHCFromProbe(kubectlArgs, ipName, clients)
+		cleanupDeps(ipName)
+	}()
+
+	testHCFromProbe(project, kubeConfigPath, lbName)
+}
+
+func setupHCFromProbe(kubectlArgs []string, ipName string, clients map[string]kubeclient.Interface) {
+	deployApp(kubectlArgs, clients, "testdata/e2e/hcFromProbe/app/")
+	replaceVariable("testdata/e2e/hcFromProbe/ingress.yaml", "\\$E2E_KUBEMCI_IP", ipName)
+}
+
+func cleanupHCFromProbe(kubectlArgs []string, ipName string, clients map[string]kubeclient.Interface) {
+	cleanupApp(kubectlArgs, clients, "testdata/e2e/hcFromProbe/app/")
+	replaceVariable("testdata/e2e/hcFromProbe/ingress.yaml", ipName, "\\$E2E_KUBEMCI_IP")
+}
+
+func testHCFromProbe(project, kubeConfigPath, lbName string) {
```

Comment:
Still todo for the test methods:
- [] break out common operations into shared function